Client/Frontend/Browser/TopTabsViewController.swift

```diff
@@ -0,0 +1,714 @@
+    static let FaderPading: CGFloat = 5
+    static let BackgroundSeparatorLinePadding: CGFloat = 5
+    static let TabTitleWidth: CGFloat = 110
+    static let TabTitlePadding: CGFloat = 10
+}
+
+protocol TopTabsDelegate: class {
+    func topTabsPressTabs()
+    func topTabsPressNewTab()
+    func topTabsPressPrivateTab()
+    func topTabsDidChangeTab()
+}
+
+protocol TopTabCellDelegate: class {
+    func tabCellDidClose(cell: TopTabCell)
+}
+
+class TopTabsViewController: UIViewController {
+    let tabManager: TabManager
+    weak var delegate: TopTabsDelegate!
+    var isPrivate = false
+    lazy var collectionView: UICollectionView = {
+        let collectionView = UICollectionView(frame: CGRectZero, collectionViewLayout: TopTabsViewLayout())
+        collectionView.registerClass(TopTabCell.self, forCellWithReuseIdentifier: TopTabCell.Identifier)
+        collectionView.showsVerticalScrollIndicator = false
+        collectionView.showsHorizontalScrollIndicator = false
+        collectionView.bounces = false
+        collectionView.clipsToBounds = false
+        
+        return collectionView
+    }()
+    
+    private lazy var tabsButton: TabsButton = {
+        let tabsButton = TabsButton()
+        tabsButton.titleLabel.text = "0"
+        tabsButton.addTarget(self, action: #selector(TopTabsViewController.tabsClicked), forControlEvents: UIControlEvents.TouchUpInside)
+        tabsButton.accessibilityIdentifier = "TopTabsView.tabsButton"
+        tabsButton.accessibilityLabel = NSLocalizedString("Show Tabs", comment: "Accessibility Label for the tabs button in the tab toolbar")
+        return tabsButton
+    }()
+    
+    private lazy var newTab: UIButton = {
+        let newTab = UIButton()
+        newTab.addTarget(self, action: #selector(TopTabsViewController.newTabClicked), forControlEvents: UIControlEvents.TouchUpInside)
+        newTab.setImage(UIImage.templateImageNamed("menu-NewTab-pbm"), forState: .Normal)
+        newTab.tintColor = UIColor(white: 0.9, alpha: 1)
+        newTab.setImage(UIImage(named: "menu-NewTab-pbm"), forState: .Highlighted)
+        newTab.accessibilityLabel = NSLocalizedString("New Tab", comment: "Accessibility label for the New Tab button in the tab toolbar.")
+        return newTab
+    }()
+    
+    private lazy var privateTab: UIButton = {
+        let privateTab = UIButton()
+        privateTab.addTarget(self, action: #selector(TopTabsViewController.privateTabClicked), forControlEvents: UIControlEvents.TouchUpInside)
+        privateTab.setImage(UIImage.templateImageNamed("menu-NewPrivateTab-pbm"), forState: .Normal)
+        privateTab.tintColor = UIColor(white: 0.9, alpha: 1)
+        privateTab.setImage(UIImage(named: "menu-NewPrivateTab-pbm"), forState: .Highlighted)
+        privateTab.accessibilityLabel = NSLocalizedString("New Tab", comment: "Accessibility label for the Private Tab button in the tab toolbar.")
+        return privateTab
+    }()
```

Comment:
Nit: For these properties, I feel like these were copied from elsewhere to replicate the style we were using in the Tabs Tray. If the code identical, I would refactor this into an `extension` class method on UIButton/TabsButton to DRY up the code so we don't have it in multiple places.